doc/examples/reconst_dki.py

```diff
@@ -48,99 +52,99 @@
+b-value. Here, we use fetch to download a multi-shell dataset which parameters
+are similar to the used on the Human Connectome Project (HCP). The total size
+of the dowloaded data is 1760 MBytes, however you only need to fetch it once.
+Parameter ``with_raw`` of function ``fetch_cenir_multib`` is set to ``False``
+to only download eddy-current/motion corrected data:
 """
 
-from dipy.data import fetch_sherbrooke_3shell
-
-fetch_sherbrooke_3shell()
+fetch_cenir_multib(with_raw=False)
 
 """
-Next, we read the saved dataset
+Next, we read the saved dataset. To decrease the influence of diffusion signal
+taylor approximation componets larger than the fourth order (componets not
+taken into account by the diffusion kurtosis tensor), we only select the
+b-values up to 2000 $s.mm^{-2}$:
 """
 
-from dipy.data import read_sherbrooke_3shell
+bvals = [200, 400, 1000, 2000]
 
-img, gtab = read_sherbrooke_3shell()
+img, gtab = read_cenir_multib(bvals)
 
 data = img.get_data()
 
+affine = img.get_affine()
+
 """
-img contains a nibabel Nifti1Image object (with the data) and gtab contains a
-GradientTable object with information about the b-values and b-vectors. The
-b-values used on the loaded dataset are visualized below
-"""
+Function ``read_cenir_multib`` return img and gtab which contains respectively
+a nibabel Nifti1Image object (where the data can be extracted) and a
+GradientTable object with information about the b-values and b-vectors.
 
-import matplotlib.pyplot as plt
+Before fitting the data, we preform some data pre-processing. We first create a
+preliminary mask and avoid pre-processing unnecessary voxels:
+"""
 
-plt.plot(gtab.bvals, label='b-values')
-fig1 = plt.gcf()
-plt.legend()
-plt.show()
-fig1.savefig('HARDI193_bvalues.png')
+maskdata, mask = median_otsu(data, 4, 2, False, vol_idx=[0, 1], dilate=1)
 
 """
-.. figure:: HARDI193_bvalues.png
-   :align: center
-   **b-values of the loaded dataset**.
-
-From the figure above, we can check that the loaded dataset contains three
-non-zero b-values as required for DKI. However the highest b-value of 3500
-$s.mm^{-2}$ is higher than normally used on DKI. Since DKI neglects diffusion
-signal components higher than the 4th order KT, a upper bound of b-value < 3000
-$s.mm^{-2}$ is normally implied to decrease the influence of higher order terms
-[Jensen2010]_. Following this, we discard the b-value shell of 3500 $s.mm^{-2}$
-before DKI fitting.
+Since the diffusion kurtosis models involves the estimation of a large number
+of parameters [TaxCMW2015]_ and since the non-Gaussian components of the
+diffusion signal are more sensitivity to artefacts [NetoHe2012]_, a fundamental
+data pre-processing step for diffusion kurtosis fitting is to denoise our data.
+For this, we use Dipy's non-local mean filter (see
+:ref:`example-denoise-nlmeans`). Note that, since the HCP-like has a large
+number of diffusion-weigthed volume, this procedure can take some minutes to
```

Comment:
How about replacing "some minutes" with "a couple of hours"? It's fine that
it takes time, considering how nice the results are following this.

On Fri, Aug 14, 2015 at 9:57 AM, Rafael Neto Henriques <
notifications@github.com> wrote:

> In doc/examples/reconst_dki.py
> <https://github.com/nipy/dipy/pull/677#discussion_r37095593>:
>
> > -   **b-values of the loaded dataset**.
> > -
> > -From the figure above, we can check that the loaded dataset contains three
> > -non-zero b-values as required for DKI. However the highest b-value of 3500
> > -$s.mm^{-2}$ is higher than normally used on DKI. Since DKI neglects diffusion
> > -signal components higher than the 4th order KT, a upper bound of b-value < 3000
> > -$s.mm^{-2}$ is normally implied to decrease the influence of higher order terms
> > -[Jensen2010]_. Following this, we discard the b-value shell of 3500 $s.mm^{-2}$
> > -before DKI fitting.
> > +Since the diffusion kurtosis models involves the estimation of a large number
> > +of parameters [TaxCMW2015]_ and since the non-Gaussian components of the
> > +diffusion signal are more sensitivity to artefacts [NetoHe2012]_, a fundamental
> > +data pre-processing step for diffusion kurtosis fitting is to denoise our data.
> > +For this, we use Dipy's non-local mean filter (see
> > +:ref:`example-denoise-nlmeans`). Note that, since the HCP-like has a large
> > +number of diffusion-weigthed volume, this procedure can take some minutes to
>
> A couple of hours =S
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/nipy/dipy/pull/677/files#r37095593>.
>